src/classes/ExpectObject_Tests.cls

```diff
@@ -0,0 +1,78 @@
+@isTest
+public class ExpectObject_Tests {
+    @isTest
+    static void ExpectEqualObjectsShouldNotThrow() {
+        Object one = new List<Account>{new Account(Name = 'Hello', Phone = '1')};
+        Object two = new List<Account>{new Account(Phone = '1', Name = 'Hello')};
+
+        Expect.that(one).shouldEqual(two);
```

Comment:
:monkey_face: This is a very unexpected result. They should not equal because there are two different lists instances here and each one has two different accounts instances. Normally this would be called equivalence but not equality.